pkg/reconciler/tenancy/initialization/apibinder/bootstrap.go

```diff
@@ -0,0 +1,276 @@
+
+// Bootstrap installs the apis that drive the apibinder initializer and starts the
+// controller once they're ready.
+func Bootstrap(ctx context.Context, cfg *rest.Config, installInto logicalcluster.Name) error {
+	klog.Infof("Beginning APISet Initializer bootstrapping in cluster %q", installInto)
+	gr := metav1.GroupResource{
+		Group:    initialization.GroupName,
+		Resource: "apisets",
+	}
+	crd, err := initializationbootstrap.CustomResourceDefinition(gr)
+	if err != nil {
+		return fmt.Errorf("could not determine CustomResourceDefinition: %w", err)
+	}
+
+	resourceSchema, err := apisv1alpha1.CRDToAPIResourceSchema(crd, invalidCharacters.ReplaceAllString(version.Get().String(), "-")) // TODO: find a sensible thing to use here
+	if err != nil {
+		return fmt.Errorf("could not convert CustomResourceDefinition to APIResourceSchema: %w", err)
+	}
+
+	scopedCfg := rest.AddUserAgent(rest.CopyConfig(cfg), "kcp-apibinder-initializer-bootstrapper")
+	kcpClusterClient, err := kcpclient.NewClusterForConfig(scopedCfg)
+	if err != nil {
+		return fmt.Errorf("could not create KCP cluster client: %w", err)
+	}
+
+	kcpClient := kcpClusterClient.Cluster(installInto)
+	apisClient := kcpClient.ApisV1alpha1()
+	resourceSchema, err = apisClient.APIResourceSchemas().Create(ctx, resourceSchema, metav1.CreateOptions{})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return fmt.Errorf("could not create APIResourceSchema: %w", err)
+	}
+	klog.Infof("Created APIResourceSchema %q|%q", installInto.String(), resourceSchema.Name)
+
+	export, err := apisClient.APIExports().Create(ctx, &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: gr.String(),
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{resourceSchema.Name},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return fmt.Errorf("could not create APIExport: %w", err)
+	}
+	klog.Infof("Created APIExport %q|%q", installInto.String(), export.Name)
+
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   2,
+		Steps:    10,
+	}, func() (done bool, err error) {
+		export, err = apisClient.APIExports().Get(ctx, export.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.Infof("Error waiting for APIExport %q|%q virtual workspace URLs to be ready: %v", installInto.String(), export.Name, err)
+			return false, err
+		}
+		return checkConditionTrueWhileLogging(export, apisv1alpha1.APIExportVirtualWorkspaceURLsReady, fmt.Sprintf("Not done waiting for APIExport %q|%q virtual workspace URLs to be ready", installInto.String(), export.Name)), nil
+	}); err != nil {
+		return fmt.Errorf("could not wait for virtual workspace URLs to be ready on APIExport: %w", err)
+	}
+	klog.Infof("APIExport %q|%q virtual workspace URLs are ready", installInto.String(), export.Name)
+
+	binding, err := apisClient.APIBindings().Create(ctx, &apisv1alpha1.APIBinding{
```

Comment:
// TODO remove creating this APIBinding when #1183 is fixed?

Reply:
Intent was for this to live in the root workspace where e.g. organization then wants to use the initializer to add e.g. clusterworkspaces